### ISSUE TITLE
Search for config files in the current directory 

### ISSUE THREAD

**Issue by daniel-brosche** [jacebrowning/gitman#152]
Hello,

I have recognized that gitman is actually not correctly working if the .gitman.yml is placed in one subdirectory and the parent directory is the git-root.
In my example I have a git repo named playground. One of the subfolders is named gitman_test. In this constellation it is not possible to use gitman.
I got the following verbose log:

```
brosche@lde275426:~/Dev/playground/gitman_test$ gitman update --verbose
[INFO    ] Updating dependencies: <all>
[INFO    ] Current directory: /home/brosche/Dev/playground/gitman_test
[INFO    ] Searching for root...
[INFO    ] Found root: /home/brosche/Dev/playground
[WARNING ] No dependencies to update
```

However it works when I copy the gitman_test to /home/brosche because there is no git root present there. Gitman would be for us more helpful without this root search because we are planing a repo structure where the gitman.yml is placed in one subdirectory.

What is the background of this search? 
Is this crucial?
In my second test as I have copied  gitman_test to /home/brosche there was no problem at all without this root.

Kind regards, Daniel...

**Comment by daniel-brosche** [comment]
I have noticed when I use -r option and tell gitman to use the current directory it works.
gitman update -r .
What do you think: To prefer the gitman.yml from the current directory makes sense?

**Comment by jacebrowning** [comment]
My thinking is that GitMan is configured per-repository (similar to submodules) and that all paths should be formed relative to the root so that dependencies can be nested.

Can you clarify **why** you want to place the configuration file somewhere other than the root of the project?

**Comment by daniel-brosche** [comment]
Well, I understand your intention in general.
We are planing something that can be considered as a embedded system / product repository. This repository contains / imports all what we need to build an embedded system.
The content of this repository consists of:
- operating system related stuff
- application related stuff

The planned structure looks something like that:
- root:
  - app
     - imports (imported app related repos by gitman)
     - .gitman.yml
     - build.sh
     - some other stuff
  - os
    - imports (imported os related repos by gitman)
    - .gitman.yml   
    - build.sh
    - some other stuff
  - build.sh

We don't want to mix the os related stuff with the app related in one gitman.yml to reduce the amount of imported repositories for those which developing only things in the app layer. Of course I see the possiblity to split os and app layer in seperate repositories and import them aggain in another top level repository but the would increase the configuration management and I think what belongs together should not be divided if not necessary.

Kind regards...

**Comment by jacebrowning** [comment]
This makes sense to me, but I have a few questions:

From which directory to you plan to call `$ gitman`?

What should happen if a configuration file is found in the current directory and the root?

Should running `$ gitman install` stop at the current directory or traverse up to the root and down through all subdirectories?

When installing nested dependencies, should GitMan look inside all subdirectories of a dependency for more GitMan configuration files?

**Comment by daniel-brosche** [comment]
> What should happen if a configuration file is found in the current directory and the root?

The configuration file of the current directory will be used.

> Should running  $ gitman install  stop at the current directory or traverse up to the root and down through all subdirectories?

The current directory is the root for the recursive dependency resolution (only top down through all subdirectories).

> When installing nested dependencies, should GitMan look inside all subdirectories of a dependency for more GitMan configuration files?

Each specified dependency could have another configuration file that will be resolved either if not explicit configured by the depth option.

> Where should  $ gitman init  generate files? (I just changed the logic in  f2a3cb2 )
In the current directory.

I'm working with a couple of tools that behaves similar to this approach. The current directory is the default location for input / output if not explicitly specified (like make -C /path/to/dir)

**Comment by jacebrowning** [comment]
When you get a chance, please provide feedback on `GitMan==1.4b1`.

**Comment by daniel-brosche** [comment]
I have checked the following scenarios from my win7 system:
	1.) config in root
		○ update from root folder
		expected behaviour: resolve dependencies -> passed
		
		○ update from subdirectory
		expected behaviour: resolve dependencies -> passed
		
		○ update from parent root folder
		expected behaviour: No dependencies to update -> passed
		
	2.) config in subfolder
		○ update from config folder 
		expected behaviour: resolve dependencies -> passed
	
		○ update from subdirectory 
		expected behaviour: resolve dependencies -> passed
	
		○ update from root directory
		expected behaviour: No dependencies to update -> passed

Overall it works :)

I have noticed that the found config path won't be logged:
```
[INFO    ] Searching for root...
[INFO    ] Found root: D:\GIT\playground
[INFO    ] Searching for config...
[INFO    ] $ mkdir D:\GIT\playground\imports
```

In my opinion your bottom up config search works. It is something that could be a little bit misleading accoding the common behaviour of other applications that I have mentioned. They are in general not that intelligent and do an automatic config search they just process the cwd or the passed folder / search folders.
In this matter the gitman could be simplified by just processing the passed config folder which is by default the cwd. Maybe I'm wrong but actually I don't see the need for the rootfolder. 
I could do the refactoring - it dosen't seems to be a big deal :)

Kind regards, Daniel...

**Comment by jacebrowning** [comment]
Thanks for the feedback. I'll consider removing root detection entirely.

**Comment by jacebrowning** [comment]
The new behavior is available in `GitMan==1.4b2`.
